File: persistence/src/main/java/io/sparkled/persistence/sequence/impl/query/GetSequenceByIdQuery.java
package io.sparkled.persistence.sequence.impl.query;

import io.sparkled.model.entity.Sequence;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.Optional;

public class GetSequenceByIdQuery implements PersistenceQuery<Optional<Sequence>> {

    private final int sequenceId;

    public GetSequenceByIdQuery(int sequenceId) {
        this.sequenceId = sequenceId;
    }

    @Override
    public Optional<Sequence> perform(QueryFactory queryFactory) {
        Sequence sequence = queryFactory
                .selectFrom(qSequence)
                .where(qSequence.id.eq(sequenceId))
                .fetchFirst();

        return Optional.ofNullable(sequence);
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/sequence/impl/query/GetSequenceByIdQuery.kt
package io.sparkled.persistence.sequence.impl.query

import io.sparkled.model.entity.QSequence.Companion.sequence
import io.sparkled.model.entity.Sequence
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetSequenceByIdQuery(private val sequenceId: Int) : PersistenceQuery<Sequence?> {

    override fun perform(queryFactory: QueryFactory): Sequence? {
        return queryFactory
            .selectFrom(sequence)
            .where(sequence.id.eq(sequenceId))
            .fetchFirst()
    }
}
